lib/matplotlib/widgets.py

```diff
@@ -905,23 +944,11 @@ def set_val(self, val):
         _api.check_shape((2,), val=val)
         val[0] = self._min_in_bounds(val[0])
         val[1] = self._max_in_bounds(val[1])
-        xy = self.poly.xy
+        xy = self._fill_verts(self.poly.xy, val)
```

Comment:
I think you can go even further

```suggestion
        self.poly.xy = self._fill_verts(self.poly.xy, val)
```